spec.html

```diff
@@ -42355,21 +42537,49 @@ <h1>
           AllocateArrayBuffer (
             _constructor_: a constructor,
             _byteLength_: a non-negative integer,
+            optional _maxByteLength_: a non-negative integer or ~empty~,
           ): either a normal completion containing an ArrayBuffer or a throw completion
         </h1>
         <dl class="header">
           <dt>description</dt>
           <dd>It is used to create an ArrayBuffer.</dd>
         </dl>
         <emu-alg>
-          1. Let _obj_ be ? OrdinaryCreateFromConstructor(_constructor_, *"%ArrayBuffer.prototype%"*, « [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] »).
+          1. Let _slots_ be « [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] »;.
+          1. If _maxByteLength_ is present and not ~empty~, let _allocatingResizableBuffer_ be *true*; otherwise let _allocatingResizableBuffer_ be *false*.
+          1. If _allocatingResizableBuffer_ is *true*, then
+            1. If _byteLength_ > _maxByteLength_, throw a *RangeError* exception.
+            1. Append [[ArrayBufferMaxByteLength]] to _slots_.
+          1. Let _obj_ be ? OrdinaryCreateFromConstructor(_constructor_, *"%ArrayBuffer.prototype%"*, _slots_).
           1. Let _block_ be ? CreateByteDataBlock(_byteLength_).
           1. Set _obj_.[[ArrayBufferData]] to _block_.
           1. Set _obj_.[[ArrayBufferByteLength]] to _byteLength_.
+          1. If _allocatingResizableBuffer_ is *true*, then
+            1. If it is not possible to create a Data Block _block_ consisting of _maxByteLength_ bytes, throw a *RangeError* exception.
+            1. NOTE: Resizable ArrayBuffers are designed to be implementable with in-place growth. Implementations may throw if, for example, virtual memory cannot be reserved up front.
+            1. Set _obj_.[[ArrayBufferMaxByteLength]] to _maxByteLength_.
           1. Return _obj_.
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-arraybufferbytelength" type="abstract operation">
+        <h1>
+          ArrayBufferByteLength (
+            _arrayBuffer_: an ArrayBuffer or SharedArrayBuffer,
+            _order_: ~SeqCst~ or ~Unordered~,
+          ): a non-negative integer
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. If IsSharedArrayBuffer(_arrayBuffer_) is *true* and _arrayBuffer_ has an [[ArrayBufferByteLengthData]] internal slot, then
+            1. Let _bufferByteLengthBlock_ be _arrayBuffer_.[[ArrayBufferByteLengthData]].
+            1. Return ℝ(GetValueFromBuffer(_bufferByteLengthBlock_, 0, ~BigUint64~, *true*, _order_)).
```

Comment:
`GetValueFromBuffer` expects its first arg to be a [Shared]ArrayBuffer,
but `_bufferByteLengthBlock_` is a Shared Data Block.
Not sure how to fix this.

Reply:
This should be a call to GetRawBytesFromSharedBlock, will fix.

Reply:
Fixed.